tensorly/backend/tensorflow_backend.py

```diff
@@ -18,270 +9,159 @@
-
-
-def truncated_svd(matrix, n_eigenvecs=None):
-    """Computes an SVD on `matrix`
-
-    Parameters
-    ----------
-    matrix : 2D-array
-    n_eigenvecs : int, optional, default is None
-        if specified, number of eigen[vectors-values] to return
-
-    Returns
-    -------
-    U : 2D-array
-        of shape (matrix.shape[0], n_eigenvecs)
-        contains the right singular vectors
-    S : 1D-array
-        of shape (n_eigenvecs, )
-        contains the singular values of `matrix`
-    V : 2D-array
-        of shape (n_eigenvecs, matrix.shape[1])
-        contains the left singular vectors
-    """
-    dim_1, dim_2 = matrix.shape
-    if dim_1 <= dim_2:
-        min_dim = dim_1
-    else:
-        min_dim = dim_2
-
-    if n_eigenvecs is None or n_eigenvecs > min_dim:
-        full_matrices = True
-    else:
-        full_matrices = False
-
-    S, U, V = tf.svd(matrix, full_matrices=full_matrices)
-    U, S, V = U[:, :n_eigenvecs], S[:n_eigenvecs], transpose(V)[:n_eigenvecs, :]
-    return U, S, V
-    
-
-def partial_svd(matrix, n_eigenvecs=None):
-    """Computes a fast partial SVD on `matrix` using NumPy
-
-        if `n_eigenvecs` is specified, sparse eigendecomposition
-        is used on either matrix.dot(matrix.T) or matrix.T.dot(matrix)
-
-        Faster for very sparse svd (n_eigenvecs small) but uses numpy/scipy
-
-    Parameters
-    ----------
-    matrix : 2D-array
-    n_eigenvecs : int, optional, default is None
-        if specified, number of eigen[vectors-values] to return
-
-    Returns
-    -------
-    U : 2D-array
-        of shape (matrix.shape[0], n_eigenvecs)
-        contains the right singular vectors
-    S : 1D-array
-        of shape (n_eigenvecs, )
-        contains the singular values of `matrix`
-    V : 2D-array
-        of shape (n_eigenvecs, matrix.shape[1])
-        contains the left singular vectors
-    """
-    ctx = context(matrix)
-    matrix = to_numpy(matrix)
-    U, S, V = numpy_backend.partial_svd(matrix, n_eigenvecs)
-    return tensor(U, **ctx), tensor(S, **ctx), tensor(V, **ctx)
-
-
-SVD_FUNS = {'numpy_svd':partial_svd, 'truncated_svd':truncated_svd}
+            full_matrices = False
+
+        S, U, V = tf.svd(matrix, full_matrices=full_matrices)
+        U, S, V = U[:, :n_eigenvecs], S[:n_eigenvecs], tf.transpose(V)[:n_eigenvecs, :]
+        return U, S, V
+
+    @property
+    def SVD_FUNS(self):
+        return {'numpy_svd': self.partial_svd,
+                'truncated_svd': self.truncated_svd}
+
+
+for name in ['float64', 'float32', 'int64', 'int32']:
+    TensorflowBackend.register_method(name, getattr(np, name))
+
+for name in ['ones', 'zeros', 'zeros_like', 'eye', 'reshape', 'transpose',
+             'where', 'sign', 'abs', 'sqrt', 'qr']:
+    TensorflowBackend.register_method(name, getattr(tf, name))
+
+for name in ['min', 'max', 'mean', 'sum', 'prod', 'all']:
+    TensorflowBackend.register_method(name, getattr(tf, 'reduce_' + name))
```

Comment:
Just a thought: we could just have a list of tuples ('source_fun', 'target_fun_name') which would also give us an easy way to document the correspondance between the wraped function and the original ones from the backend, should we want to.